amd_gpu/gpu.c

```diff
@@ -163,10 +163,12 @@ const char* err_to_str(cl_int ret)
 		return "CL_INVALID_LINKER_OPTIONS";
 	case CL_INVALID_DEVICE_PARTITION_COUNT:
 		return "CL_INVALID_DEVICE_PARTITION_COUNT";
+#ifndef __APPLE__
```

Comment:
This guard is not only a fix for APPLE it fixes also `ubuntu 14.04`. My suggestion is to check the AMD APP SDK VERSION and disable this code for old APP SDK's